app/controllers/admin/orders_controller.rb

```diff
@@ -4,13 +4,24 @@ class OrdersController < AdminController
     def change_state
       @order = Order.find(params[:order_id])
 
-      unless @order.make_canceled
-        flash[:error] = "Order cannot be canceled."
+      if @order.make_canceled
+        flash[:success] = "Order successfully canceled."
+      else
+        flash[:error] = "This order cannot be canceled."
```

Comment:
why? there would be some validation messages I assume you could show to the user?